go-app.js

```diff
@@ -127,6 +127,21 @@ go.app = function() {
         choices: self.im.config.localities.map(function (locality) {
           return new Choice(locality, locality);
         }),
+        next: 'Landmark_Entry'
+      });
+    });
+
+    self.states.add('Landmark_Entry', function(name) {
+      var question = $("What is the closest landmark for the patient?");
+      var error = $("Please enter a landmark");
+      return new FreeText(name, {
+        question: question,
+        check: function(content) {
+          if (content.length === 0) {
+            return error;
+          }
```

Comment:
For the FreeText state you need to do what you did before to return an
error message, but you are never going to have content length 0 (because
the interaction machine won't pass you empty content).